src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,167 @@
+            last = prev;
+        } else {
+
+            if (prev == null) {
+                next.previous = null;
+                first = next;
+            } else {
+                prev.next = next;
+                next.previous = prev;
+            }
+        }
+    }
+
     @Override
-    public void add(T value) {
+    public void add(T element) {
+        Node<T> newNode = new Node<>(element);
+        if (first == null) {
+            newNode.next = null;
+            newNode.previous = null;
+            first = newNode;
+            last = newNode;
+        } else {
+            last.next = newNode;
+            newNode.previous = last;
+            last = newNode;
+        }
+        size++;
     }
 
     @Override
-    public void add(T value, int index) {
+    public void add(T element, int index) {
+        size++;
+        isIndexOk(index);
+        size--;
+        Node<T> newNode = new Node(element);
+        if (index == size) {
+            add(element);
+        } else if (index == 0) {
+            Node<T> oldNode = first;
+            oldNode.previous = newNode;
+            newNode.next = oldNode;
+            first = newNode;
+            size++;
+        } else {
+            Node<T> oldNode = getNode(index);
+            Node<T> oldPrevious = oldNode.previous;
+            oldPrevious.next = newNode;
+            oldNode.previous = newNode;
+            newNode.previous = oldPrevious;
+            newNode.next = oldNode;
+            size++;
+        }
+
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        isIndexOk(index);
+        Node<T> result = getNode(index);
+        return result.element;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        isIndexOk(index);
+        Node<T> node = first;
+        int count = 0;
+        while (node.next != null) {
+            if (count == index) {
+                T oldValue = node.element;
+                node.element = value;
+                return oldValue;
+            }
+            count++;
+            node = node.next;
+        }
+        return node.element;
+    }
+
+    @Override
+    public boolean isEmpty() {
+        return size == 0;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        isIndexOk(index);
+        Node<T> result = getNode(index);
+        T element = result.element;
+        unLink(result);
+        size--;
+        return element;
     }
 
     @Override
-    public boolean remove(T object) {
+    public boolean remove(T value) {
+        Node<T> node = first;
+        for (int i = 0; i < size; i++) {
+            if ((value == null && node.element == null) || value != null && node.element != null
+                    && node.element.equals(value)) {
```

Comment:
condition may be simplified:
```suggestion
            if (value == node.element || value != null && value.equals(node.element)) {
```